test/plots/waterfallPlotTests.ts

```diff
@@ -1,123 +1,251 @@
+        });
+        plot.destroy();
+        svg.remove();
+      });
+
+      it("bars placed at current sum", () => {
+        let bars = plot.content().selectAll("rect.waterfall-decline");
+        assert.strictEqual(bars.size(), declineBarData.length - 1, "all bars are decline except for first");
+        let yAccessor = plot.y().accessor;
+        let sum = 0;
+        bars.each(function(d, i) {
+          let dataY = yAccessor(d, i, dataset);
+          let bar = d3.select(this);
+          assert.closeTo(numAttr(bar, "y"), yScale.scale(sum),
+            window.Pixel_CloseTo_Requirement, "growth bar top at previous sum");
+          sum += dataY;
+          assert.closeTo(numAttr(bar, "y") + numAttr(bar, "height"), yScale.scale(sum),
+            window.Pixel_CloseTo_Requirement, "growth bar bottom at final sum");
+        });
+
+        plot.destroy();
+        svg.remove();
+      });
     });
 
-    it("bars are classed correctly", () => {
-      let bars = renderArea.selectAll("rect")[0];
-      let data = dataset.data();
-      let totalAccessor = plot.total().accessor;
-      bars.forEach((bar, index) => {
-        let selection = d3.select(bar);
-        let isTotal = totalAccessor(data[index], index, dataset);
-        if (isTotal) {
-          assert.isTrue(selection.classed("waterfall-total"));
-        } else {
-          let yAccessor = plot.y().accessor;
-          if (yAccessor(data[index], index, dataset) > 0) {
-            assert.isTrue(selection.classed("waterfall-growth"));
+    describe("denoting total bars", () => {
+      let svg: d3.Selection<void>;
+      let dataset: Plottable.Dataset;
+      let xScale: Plottable.Scales.Category;
+      let yScale: Plottable.Scales.Linear;
+      let plot: Plottable.Plots.Waterfall<string, number>;
+      let SVG_WIDTH = 600;
+      let SVG_HEIGHT = 400;
+      let data = [
+        { x: "A", y: 20, t: true },
+        { x: "B", y: 5, t: false },
+        { x: "C", y: 25, t: true },
+        { x: "D", y: -10, t: false },
+        { x: "E", y: 15, t: true }
+      ];
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        dataset = new Plottable.Dataset(data);
+        xScale = new Plottable.Scales.Category();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Waterfall<string, number>();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+      });
+
+      it("can set the total property", () => {
+        let accessor = (d: any) => d.t;
+        assert.strictEqual(plot.total(accessor), plot, "setter returns calling object");
+
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+        let bars = plot.content().selectAll("rect");
+        let totalBarIndices = data.map((d, i) => i).filter((index) => accessor(data[index]));
+        totalBarIndices.forEach((totalBarIndex) => {
+          let totalBar = d3.select(bars[0][totalBarIndex]);
```

Comment:
I believe this way to select element is discouraged in best practice
also, why don't you just iterate through all the bars and check that it is classed or not classed based on `t`?